src/create_agent/init_agent.jl

```diff
@@ -134,6 +134,27 @@ function init_agent(
         agent.states["action"] = missing
     end
 
+    
+
+    #Go through each specified shared parameter
+    for (shared_parameter_key, dict_value) in shared_parameters
+        #Unpack the shared parameter value and the derived parameters
+        (shared_parameter_value, derived_parameters) = dict_value
+
+        agent.shared_parameters[shared_parameter_key] = SharedParameter(
+            value = shared_parameter_value,
+            derived_parameters = derived_parameters,
+        )
+        #Check agent for settings of shared parameters
+        check_agent(agent)
```

Comment:
Ikke check agent inde i det her for loop. Vent med det til du er færdig med at lave den. Du kan gøre det helt til sidst i funktionen, ca.